megazords/ios-rust/build-xcframework.sh

```diff
@@ -146,6 +146,7 @@ cp "$WORKING_DIR/$FRAMEWORK_NAME.h" "$COMMON/Headers"
 cp "$REPO_ROOT/components/rc_log/ios/RustLogFFI.h" "$COMMON/Headers"
 cp "$REPO_ROOT/components/viaduct/ios/RustViaductFFI.h" "$COMMON/Headers"
 $CARGO uniffi-bindgen generate "$REPO_ROOT/components/nimbus/src/nimbus.udl" -l swift -o "$COMMON/Headers"
+$CARGO uniffi-bindgen generate "$REPO_ROOT/components/external/glean/glean-core/src/glean.udl" -l swift -o "$COMMON/Headers"
```

Comment:
hmmm we'll need to be extra careful here to make sure the `uniffi_bindgen` version glean uses to generate its scaffolding is compatible with the version we use here. This also means that we are essentially tying our `uniffi` version in `application-services` with the `uniffi` version used in glean (at least in the short term, until we get the `uniffi` versioning work done.

Versions being tied doesn't sit well with me, but I can't see a way around it for now/

So the consequences of this are:
-  Whenever we bump the glean submodule (and it includes an updated `uniffi`) the iOS builds will fail, and CI will catch them forcing us to either align the version here in application-services, or go back and align the version in glean and cutting another release there
- We will **not** be able to bump uniffi in this repository on its own. Whenever we want to bump uniffi here, we'll have to first bump it in glean, cut a release there, then update it here

I'll try to spend a bit of time thinking about this so we can move without blocking this work
cc @mhammond and @bendk

Reply:
I wanted to point out that I chatted briefly with @bendk about this, and with the upcoming changes in https://github.com/mozilla/uniffi-rs/pull/1271 I'm less concerned about the coupling between the versions. That PR changes compatibility to be less about the uniffi version and more about the low-level contract between the bindings and the scaffolding (which we don't change often at all)

And since the UDL we use is the same one from the submodule, there shouldn't be any incompatibility there.

I won't block on this anymore, and even if this PR lands first - the uniffi naming changes are already approved/agreed upon so it'll catch up quickly 🚀

Reply:
I think A-S and Glean are likely going to both be using a fairly recent version of uniffi most of the time, so hopefully the compatibility problems will be minimal. Since there are already changes proposed that will make this even easier, I'm okay with keeping Glean on the same version of A-S, and I'm sure we can work it out if we need to differ for some reason.